### PR TITLE
Provide hooks to override the get_symbol runtime calls when jitting

### PR DESCRIPTION
**Pull request by steven-johnson**
At present these are infeasible to customize when jitting, and the
exisitng set_jit_externs() approach isn’t currently workable (or easily
made workable), as discussed in PR#1725. This takes a more brute-force
approach, using the same technique already used in JITModule.

### CODE REVIEW

File: test/correctness/load_library.cpp [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    handlers.custom_load_library = my_load_library_impl;
+    handlers.custom_get_library_symbol = my_get_library_symbol_impl;
+    Internal::JITSharedRuntime::set_default_handlers(handlers);
+
+    Var x("x");
+    Func f("f");
+
+    f(x) = cast<int32_t>(x);
+
+    Target target = get_jit_target_from_environment().with_feature(Target::HVX_64);
```
Comment (dsharletg) [inline test/correctness/load_library.cpp]: This test should probably be guarded by a WITH_HEXAGON check, or something like that (I think this test will fail if LLVM is built without Hexagon support).

Comment (steven-johnson) [inline test/correctness/load_library.cpp]: Ouch, yes.

**Comment by steven-johnson** [comment]
PTAL

File: test/correctness/load_library.cpp [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+// feature in LLVM (unlike, say, Hexagon).
+
+namespace {
+
+void my_error_handler(void* u, const char *msg) {
+    if (!strstr(msg, "OpenCL API not found")) {
+        fprintf(stderr, "Saw unexpected error: %s\n", msg);
+        exit(-1);
+    }
+    printf("Saw expected error: %s\n", msg);
```
Comment (dsharletg) [inline test/correctness/load_library.cpp]: Wouldn't this test "pass" if the custom JIT handlers were broken, and the OpenCL runtime was also missing?

Comment (steven-johnson) [inline test/correctness/load_library.cpp]: Indeed you are correct. Fixed.

**Pull request merged**